M2/Macaulay2/d/pthread.d

```diff
@@ -218,6 +220,16 @@ setupfun("setIOSynchronized",setIOSynchronized);
 setupfun("setIOUnSynchronized",setIOUnSynchronized);
 setupfun("setIOExclusive",setIOExclusive);
 
+export getIOThreadMode(e:Expr):Expr := (
```

Comment:
How can I use this info at the top-level?

Reply:
This returns 0, 1, or 2, depending on whether it's unsynchronized, synchronized, or exclusive, respectively.  Previously, 0 was the default, but in this PR, the proposed default is 1.

Here's an example:

```m2
i1 : getIOThreadMode() -- the new default, try to synchronize how threads print

o1 = 1

i2 : scan(20, i -> schedule(() -> print "Hello, world!"))
Hello, world!
Hello, world!
Hello, world!
Hello, world!
Hello, world!
Hello, world!

Hello, world!
i3 : Hello, world!
Hello, world!
Hello, world!Hello, world!
Hello, world!
Hello, world!
Hello, world!
Hello, world!

Hello, world!Hello, world!

Hello, world!Hello, world!

Hello, world!
setIOExclusive() -- only the main thread can print

i4 : getIOThreadMode()

o4 = 2

i5 : scan(20, i -> schedule(() -> print "Hello, world!")) -- notice nothing is printing (yet)

i6 : setIOUnSynchronized() -- go crazy!  the current default, which leads to fun things like this:

i7 : Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
Hello, world!Hello, world!
ello, world!

internal error: beginning of line marker not within buffer
internal error: beginning of line marker not within buffer
internal error: beginning of line marker not within buffer
internal error: beginning of line marker not within buffer
Aborted (core dumped)

Process M2 exited abnormally with code 134
```
As Dave mentioned above, this is all a mess and should be scrapped eventually.  But I figured it would be nice to at least have a function that lets us find out the current thread mode for the time being.

Reply:
Thanks! I see, I didn't know you could change the thread mode this at the top level. I agree that the whole printing system is not really suited for threads.